src/kvstore/raftex/Host.cpp

```diff
@@ -401,14 +402,14 @@ Host::prepareAppendLogRequest() {
             LOG(INFO) << idStr_ << "Can't find log " << lastLogIdSent_ + 1
                       << " in wal, send the snapshot";
             sendingSnapshot_ = true;
-            part_->snapshot_->sendSnapshot(part_, addr_).then([this] (Status&& status) {
+            part_->snapshot_->sendSnapshot(part_, shared_from_this(), addr_)
+                .then([idStr = this->idStr_] (Status&& status) {
                 if (status.ok()) {
-                    LOG(INFO) << idStr_ << "Send snapshot succeeded!";
+                    LOG(INFO) << idStr << "Send snapshot succeeded!";
                 } else {
-                    LOG(INFO) << idStr_ << "Send snapshot failed!";
+                    LOG(INFO) << idStr << "Send snapshot failed!";
                     // TODO(heng): we should tell the follower i am failed.
                 }
-                sendingSnapshot_ = false;
```

Comment:
why delete this logic?

Reply:
It is moved into SnapshotManager, during sending snapshot, the Host maybe destructed.